src/unstract/sdk/llm.py

```diff
@@ -109,7 +109,7 @@ def complete(
                     process_text_output = {}
             return {LLM.RESPONSE: response, **process_text_output}
         except Exception as e:
-            raise parse_llm_err(e) from e
+            raise parse_llm_err(e, self._llm_instance) from e
```

Comment:
Wondering if we can have an approach where we do
self.parse_error()  # self will point to an Unstract LLM Adapter instance for example
-> Anthropic.parse_error()
   -> here we parse Anthropic specific errors alone

The approach we have taken now is to have parse functions at the LLM level where we check for the exception type and then route to individual Unstract Adapter type's parse() functions.

Instead if we use botton up approach where we define parse functions at the Unstract adapter type's implementation and make sure that based on the llm instance used, the call will land to the specific parse() ?

We can have a discussion around this if the msg is not clear.